### PR TITLE
Make setValue work with an AnimatedValue as input on Web

### PR DESCRIPTION
**Pull request by jfrolich**

### CODE REVIEW

**Comment by ianmartorell** [comment]
What kind of statement wasn't working before this pull request?

**Comment by jfrolich** [comment]
`value.setValue(new Animated.Value(1))`

File: src/core/AnimatedValue.js [hunk 1]
```diff
@@ -10,7 +10,11 @@ export default class AnimatedValue extends InternalAnimatedValue {
   setValue(value) {
     this.__detachAnimation(this._animation);
     if (Platform.OS === 'web') {
-      this._updateValue(value);
+      if (typeof value === "number") {
```
Comment (EvanBacon) [inline src/core/AnimatedValue.js:13]: Rewrite this whole statement:
```js
if (typeof value === "number") {
  if (ReanimatedModule.setValue) {
    // FIXME Remove it after some time
    // For OTA-safety
    // FIXME handle setting value with a node
    ReanimatedModule.setValue(this.__nodeID, value);
  } else {
    this._updateValue(value);
  }
} else {
  evaluateOnce(set(this, value), this);
}
```

**Comment by jakub-gonet** [comment]
I tried testing it on the web and it froze, probably due to `evaluateOnce` or `set`. Prepared #1088 as fix.

**Pull request closed without merging**